h/views/api/group_members.py

```diff
@@ -99,9 +100,22 @@ def add_member(context: GroupMembershipContext, request):
     if context.user.authority != context.group.authority:
         raise HTTPNotFound()
 
+    if request.body:
+        appstruct = EditGroupMembershipAPISchema().validate(json_payload(request))
+        roles = appstruct["roles"]
+    else:
+        # This doesn't mean the membership will be created with no roles:
+        # default roles will be applied by the service.
+        roles = None
```

Comment:
For backwards-compatibility add-group-membership requests with no body are still allowed and they add memberships with the default role.